File: python/leetcode/271.py
class Codec:
    def encode(self, strs):
        res = ""
        for s in strs:
            res += str(len(s)) + '#' + s
        return res

    def decode(self, s):
        i, res = 0, []
        while i < len(s):
            idx = s.find('#', i)
            size = int(s[i:idx])
            res.append(s[idx+1:idx+1+size])
            i = idx + size + 1
        return res

File: cpp/leetcode/271.cpp
class Codec {
public:
    string encode(vector<string>& strs) {
        string res = "";
        for(auto str : strs) res += to_string(str.size()) + "#" + str;
        return res;
    }

    vector<string> decode(string s) {
        vector<string> res;
        for(int i = 0; i < s.size(); ){
            int idx = s.find_first_of('#', i);
            int len = stoi(s.substr(i, idx - i));
            res.push_back(s.substr(idx + 1, len));
            i = idx + len + 1;
        }
        return res;
    }
};